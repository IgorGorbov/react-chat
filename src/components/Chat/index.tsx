import * as React from 'react';

import styled from 'styled-components';

import SectionContacts from '../SectionContacts';
import SectionContactHeader from '../SectionContactHeader';
import SectionMessagesList from '../../containers/SectionMessagesList';
import SectionUserNewMessage from '../../containers/SectionNewMessage';

export const GridLayout = styled.div`
  display: grid;
  margin: 80px auto 0 auto;
  width: 75%;
  grid-template-columns: 1fr 3fr;
  grid-auto-rows: minmax(20px, auto);
  background-color: #485f78;
  box-shadow: 0 10px 20px rgba(0, 0, 0, 0.19), 0 6px 6px rgba(0, 0, 0, 0.23);
`;

const Chat = (): JSX.Element => {
  return (
    <GridLayout>
      <SectionContacts />
      <SectionContactHeader />
      <SectionMessagesList />
      <SectionUserNewMessage />
    </GridLayout>
  );
};

export default Chat;
