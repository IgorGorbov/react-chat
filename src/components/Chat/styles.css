.container {
    display: grid;
    margin: 80px auto 0 auto;
    width: 75%;
    grid-template-columns: 1fr 3fr;
    grid-auto-rows: minmax(20px, auto);
    background-color: #485F78;
    box-shadow: 0 10px 20px rgba(0,0,0,0.19), 0 6px 6px rgba(0,0,0,0.23);
}
.contactsList {
    display: flex;
    position: relative;
    flex-direction: column;
    grid-column: 1;
    grid-row-start: 1;
    grid-row-end: 4;
    background-color: #2F3E4E;
}

.contactHeader {
    grid-column: 2;
    grid-row: 1;
    background-color: beige;
}

.messagesList {
    max-height: 550px;
    overflow-y: scroll;
    grid-column: 2;
    grid-row: 2;
    background-color: #E7EAEA;
}

.inputField {
    grid-column: 2;
    grid-row: 3;
    max-height: 40px;
    width: 100%;
    display: flex;
    padding: 0;
    margin: 0;
    background-color: #FFFFFF;
}

.userWrapper {
    display: flex;
    flex-wrap: wrap;
    padding: 20px 0 20px 30px;
}

.userAvatar {
    max-height: 80px;
    border: 4px solid green;
    cursor: pointer;
}

.userName {
    color: white;
    font-size: 18px;
    margin-left: 10px;
    align-self: center;
    font-family: Titillium Web, sans-serif;
}

.inputSearch {
    width: 100%;
    padding: 0 1px;
}

.buttonControls {
    display: flex;
    width: 100%;
    margin: auto 0 0 0;
    align-self: center;
    padding: 0;
}
.buttonControls button {
    width: 50.5%;
    margin-right: -4px !important;
}

.inputSearch input {
    color: white !important;
    background-color: #485F78 !important;
}

.contactHeader {
    display: flex;
    padding-left: 20px;
    height: 90px;
    max-height: 110px;
    background-color: #fff;
}

.contactHeaderAvatar {
    align-self: center;
    max-width: 65px !important;
    max-height: 65px;
    border: 2px solid gray;
}

.contactUserInfo {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}
.contactUserName {
    margin-left: 10px;
    align-self: center;
    color: #000;
    font-size: 20px;
    font-weight: 500;
    font-family: Titillium Web, sans-serif;
}

.contactUserStatus {
    color: green;
}

.userDeleteButton {
    align-self: center;
    width: 60px;
    height: 50px;
    margin: 0 10px 0 auto !important;
}
.textInputMessage {
    width: 92%;
}

.ButtonSend {
    width: 8%;
    margin-right: -4px !important;
}
