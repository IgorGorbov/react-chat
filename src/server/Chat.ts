const initialData = require('../chats');

module.exports = function() {
  const chats = new Map(initialData);

  return chats;
};
