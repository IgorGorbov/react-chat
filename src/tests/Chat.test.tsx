import * as React from "react";
import toJSON from 'enzyme-to-json';
import { shallow } from "enzyme";
import Chat from "../components/Chat/index";

describe('testing Chat component', () => {

  it("should contain GridLayout", () => {
    const result = shallow(<Chat/>);
    expect(toJSON(result).type).toEqual('styled.div')
  });

  it("should contain SectionContacts", () => {
    const result = shallow(<Chat/>);
    expect(result.find('SectionContacts')).toHaveLength(1)
  });

  it("should contain SectionContactHeader", () => {
    const result = shallow(<Chat/>);
    expect(result.find('SectionContactHeader')).toHaveLength(1)
  });

  it("should contain SectionMessagesList", () => {
    const result = shallow(<Chat/>);
    expect(result.find('SectionMessagesList')).toHaveLength(1)
  });

  it("should contain SectionUserNewMessage", () => {
    const result = shallow(<Chat/>);
    expect(result.find('SectionUserNewMessage')).toHaveLength(1)
  });

  it('snapshot Chat component', () => {
    const wrapper = shallow(<Chat />);
    const tree = toJSON(wrapper);
    expect(tree).toMatchSnapshot();
  })
});
