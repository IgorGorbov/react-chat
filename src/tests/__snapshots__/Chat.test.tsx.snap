// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Chat component snapshot Chat component 1`] = `
<styled.div>
  <SectionContacts />
  <SectionContactHeader />
  <SectionMessagesList />
  <SectionUserNewMessage />
</styled.div>
`;
